### PR TITLE
 update.nix: Run update scripts in parallel

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
`nix-shell maintainers/scripts/update.nix --argstr path gnome3` is just too slow the old way.

cc @garbas @matthewbauer @hedning 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
Somewhat related, in #50983 I implement support in case of Python packages.

File: maintainers/scripts/update.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import concurrent.futures
+import json
+import os
+import subprocess
+import sys
+
+def run_update_script(package):
+    print(f" - {package['name']}: UPDATING ...", file=sys.stderr)
+
+    subprocess.run(package['updateScript'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
```
Comment (FRidh) [inline maintainers/scripts/update.py]:
```suggestion
    subprocess.run(package['updateScript'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True, shell=True)
```
Currently it is possible to not just pass in a script, but also arguments.

Comment (FRidh) [reply maintainers/scripts/update.py]: Note this solved the issue I had.

Comment (jtojnar) [reply maintainers/scripts/update.py]: Would not it be better to add shebang to the `update-python` script?

Comment (FRidh) [reply maintainers/scripts/update.py]: That's another way. In any case, the interface needs to be documented.

Comment (FRidh) [reply maintainers/scripts/update.py]: Just to be clear, I am not saying we should have it with arguments. If we need to create a derivation for every update we may as well have arguments in there.

Comment (jtojnar) [reply maintainers/scripts/update.py]: Allowing lists will allow us to avoid needing derivations per package.

**Comment by jtojnar** [comment]
Added the docs and extra args to `update.nix`.

**Comment by jtojnar** [comment]
One thing I do not like about this is that `KeyboardInterrupt` does not work very well with threads.

**Review by FRidh — APPROVED** [review]

**Review by FRidh — DISMISSED** [review]
Please update the commit message describing the relevant changes:
- runs in parallel
Change in semantics:
- update script needs to be an executable
- updateScript is a list

**Review by FRidh — APPROVED** [review]

**Pull request merged**